_overviews/scala3-migration/incompat-type-inference.md

```diff
@@ -68,19 +68,30 @@ class Child extends Parent {
 Scala 2 reflective calls are dropped and replaced by the broader [Programmatic Structural Types]({{ site.scala3ref }}/changed-features/structural-types.html).
 
 Scala 3 can imitate Scala 2 reflective calls by making `scala.reflect.Selectable.reflectiveSelectable` available wherever `scala.language.reflectiveCalls` is imported.
-However the Scala 3 compiler does not infer structural types by default, and thus fails at compiling:
 
+{% tabs define_structural %}
+{% tab 'Scala 2 and 3' %}
 ```scala
 import scala.language.reflectiveCalls
 
 val foo = new {
   def bar: Unit = ???
 }
+```
+{% endtab %}
+{% endtabs %}
 
+However the Scala 3 compiler does not infer structural types by default, and so infers type `Object` for `foo`, rather than `{ def bar: Unit }` as in Scala 2.13, so the following structural selection fail to compile:
```

Comment:
```suggestion
However the Scala 3 compiler does not infer structural types by default.
It infers the type `Object` for `foo` instead of `{ def bar: Unit }`.
Therefore, the following structural selection fails to compile:
```